Add a very simple example program to the README

lol768 opened the pull request:
This PR proposes a new example be added to the README that can:

* Easily be compiled
* Will work without modifications if compiled and run (assuming server exists etc etc)
* Showcases building the client, adding a channel, sending a message, listening for events etc

Please point out any issues you can see with the example code.

mbax commented:
Basing on your initial design, in a way easier to view (not markdown) and for me to handle PRs on: 

https://github.com/KittehOrg/KICLReadmeExample/blob/master/Example.java

lol768 commented:
\<@mbaxter\> I suppose the example class isn't going to get any more attention currently, may as well update PR and merge it in

Updated to use new code and rebased.

lol768 commented:
Neat, but it should ignore PRs which don't change *.java files
